subsys/bluetooth/host/conn.c

```diff
@@ -2232,7 +2232,10 @@ static void deferred_work(struct k_work *work)
 		 * auto connect flag if it was set, instead just cancel
 		 * connection directly
 		 */
-		bt_le_create_conn_cancel();
+		if (bt_le_create_conn_cancel() == -ENOBUFS) {
```

Comment:
How can we guarantee that in a more complex scenario this won't fail? We simply can't. Ideally, this should be completely refactored. But it is too complex task. Therefore, I suggest to print warning here if no buffers available and increase `CONFIG_BT_BUF_CMD_TX_COUNT` in your application.

Reply:
> So instead of attempting to correct the problem, the solution is to do nothing?

The solution is to refactor this such that it doesn't fails. This just postpones the problem. Who can't guarantee that in this 10ms no other call to Host (or initiated by Host itself) won't eat freed buffer?

Reply:
> Won't it simply retry in that case?

It needs at least to stop, otherwise it can run forever and not clear what will happen then.

Reply:
> It needs at least to stop, otherwise it can run forever and not clear what will happen then.

If it retries forever, then something went very wrong :D It could be possible to add a max number of retries before asserting or something